src/lesson2-homework.ts

```diff
@@ -1,129 +1,264 @@
+  function excercise9() {
+    const logger = localLogger('excercise9');
+    // TODO: declare a `Book` class with a constructor and a method `getInfo` which returns the book info as a string
+    // TODO: constructor should take three parameters - title and year of publication
+    // TODO: method `getInfo` should return the book title and year as a string
+    // TODO: create a book object and call the method `getInfo`, print the result to console
+    // TODO: assign a new value to the year property
+    // TODO: call the method `getInfo` again
+    // TODO: add a new method `getAge` which returns the age of the book (current year - year of publication)
+    // TODO: call the method `getAge` and print the result to console
+    // TODO: add a new method `revise` which takes a new year as a parameter and updates the year property, add validation to the method - year can not be in the future, year can not be less than prev year
+    // TODO: call the method `revise` and pass a new year as a parameter
+    // TODO: add private modifier to the year property
+    // TODO: try to access the year property from outside of the class - observe the error
+    // TODO: change protected modifier to the year property, remove private modifier
+    // TODO: create a subclass `Magazine` which extends `Book` class
+    // TODO: add a new properties `month` and `day` to the `Magazine` class
+    // TODO: add constructor override to the Magazine class which takes four parameters - title, year, month and day
+    // TODO: use super keyword to call the `Book` class constructor with title and year parameters
+    // TODO: add a method override `getInfo` to the `Magazine` class which prints the magazine info to console
+    // TODO: use super keyword to call the `getInfo` method of the `Book` class
+    // TODO: create a magazine object and call the method `getInfo`, print the result to console
+    // TODO: call the inherited method `getAge` of the magazine object and print the result to console
+    class Book {
+      protected title: string;
+      protected year: number;
+      constructor(title: string, year: number) {
+        this.title = title;
+        this.year = year;
+      }
+      getInfo() {
+        return `${this.title}, year ${this.year}`;
+      }
+      getAge() {
+        const currentYear: number = new Date().getFullYear();
+        return currentYear - this.year;
+      }
+      revise(newYear: number) {
+        const currentYear: number = new Date().getFullYear();
+        if (newYear >= this.year && newYear <= currentYear) {
+          this.year = newYear;
+        } else {
+          logger(`${newYear} not in range`);
+        }
+      }
+    }
+
+    class Magazine extends Book {
+      protected month: number;
+      protected day: number;
+      constructor(title: string, year: number, month: number, day: number) {
+        super(title, year);
+        this.month = month;
+        this.day = day;
+      }
+      getInfo() {
```

Comment:
method getInfo should have a proper return type and use super to call the inherited getInfo method. Check TODO comment in the exercise